lib/ansible/modules/systemd.py

```diff
@@ -400,6 +400,17 @@ def main():
                 # Check for loading error
                 if is_systemd and not is_masked and 'LoadError' in result['status']:
                     module.fail_json(msg="Error loading unit file '%s': %s" % (unit, result['status']['LoadError']))
+
+        # Workaround for https://github.com/ansible/ansible/issues/71528
+        elif err and rc == 1 and 'Failed to parse bus message' in err:
+            result['status'] = parse_systemctl_show(to_native(out).split('\n'))
+
+            (rc, out, err) = module.run_command("{systemctl} list-units '{unit}*'".format(systemctl=systemctl, unit=unit))
```

Comment:
@samdoran I believe you should use `systemctl list-units --all` . Without `--all` flag the `inactive (dead)` services will be not listed. Plz see my comment for more info https://github.com/ansible/ansible/issues/71528#issuecomment-722701620

```
root@Ubuntu18:/# systemctl list-units actions.runner.monolithprojects-ansible-github_actions_runne.Ubuntu18.service
0 loaded units listed. Pass --all to see loaded but inactive units, too.
To show all installed unit files use 'systemctl list-unit-files'.

root@Ubuntu18:/# systemctl list-units --all actions.runner.monolithprojects-ansible-github_actions_runne.Ubuntu18.service
UNIT                                                                          LOAD   ACTIVE   SUB  DESCRIPTION                                                                             
actions.runner.monolithprojects-ansible-github_actions_runne.Ubuntu18.service loaded inactive dead GitHub Actions Runner (monolithprojects-ansible-github_actions_runner-testrepo.Ubuntu18)

1 loaded units listed.
To show all installed unit files use 'systemctl list-unit-files'.
```

Reply:
Have you tried with #72363? I believe `list-unit-files` should take care of this issue.

Reply:
Ah sorry, did not notice that open PR. Yes `list-unit-files` works too. Thanks.

Reply:
Ok, that's great news. Thank you for confirming. I was hoping I didn't have to fix this _yet again_. :) I have opened backport PRs for 2.9 and 2.10, so this fix will in the next `.z` release of Ansible 2.9 and 2.10.